src/Symfony/Component/Routing/RouteCompiler.php

```diff
@@ -198,6 +206,10 @@ private static function compilePattern(Route $route, string $pattern, bool $isHo
             $tokens[] = ['text', substr($pattern, $pos)];
         }
 
+        if ($fixStaticVariables) {
+            $tokens = self::mergeContiguousTextTokens($tokens);
```

Comment:
Thre is a critical thing missing: the static value must be set as the default of the corresponding attribute.
Please add a few test cases that cover the need.